src/property.h

```diff
@@ -141,6 +142,63 @@ class FntProperties : public Properties {
   public:
 };
 
+/** Iterator on Properties
+ * @note it assumes that items are stored in an increasing order of their keys.
+ * For reading, items are retrieved in the same order. Each item is retrieved
+ * once.
+ */
+class PropsIterator {
+  public:
+    PropsIterator(const Properties& props)
+    {
+        index = 0;
+        m_key = 0;
+        m_val = 0;
+
+        keys = props.get_keys();
+        values = props.get_values();
+        items_nb = keys.size();
+
+        assert(keys.size() == values.size());
+    }
+
+    inline const size_t& curr_loc() const
+    {
+        return (index < items_nb) ? keys.at(index) : m_key;
+    }
+
+    inline const uint32_t& curr_mark() const
+    {
+        return (index < items_nb) ? values.at(index) : m_val;
+    }
+
+    inline bool is_marked(const size_t loc) const
+    {
+        return (index < items_nb) ? (keys.at(index) == loc) : false;
+    }
+
+    inline bool next() const
+    {
+        index++;
+        return (index < items_nb);
+    }
+
+    inline bool in_range(const size_t min, const size_t max) const
+    {
+        return (index < items_nb) ? (keys[index] >= min && keys[index] < max)
+                                  : false;
+    }
+
+  private:
+    std::vector<size_t> keys;
+    std::vector<uint32_t> values;
+
+    mutable size_t index;
+    size_t items_nb;
+    size_t m_key;
```

Comment:
What are `m_key` and `m_val`? If they are default values, then you should call them `m_default_val` and `m_default_key`.

Reply:
👍 

- [ ] `m_default_val`  `m_default_key`